sklearn/tree/_splitter.pyx

```diff
@@ -240,25 +236,17 @@ cdef class Splitter:
 cdef class BaseDenseSplitter(Splitter):
     cdef const DTYPE_t[:, :] X
 
-    cdef np.ndarray X_idx_sorted
-    cdef INT32_t* X_idx_sorted_ptr
-    cdef SIZE_t X_idx_sorted_stride
     cdef SIZE_t n_total_samples
-    cdef SIZE_t* sample_mask
 
     def __cinit__(self, Criterion criterion, SIZE_t max_features,
                   SIZE_t min_samples_leaf, double min_weight_leaf,
                   object random_state):
-
-        self.X_idx_sorted_ptr = NULL
-        self.X_idx_sorted_stride = 0
-        self.sample_mask = NULL
+        pass
```

Comment:
Is `__cinit__` needed anymore in this case?

Reply:
I do not think so. There are no attributes to initialize in the `__cinit__` method.